csrc/host_ir/allocate_and_deallocate.h

```diff
@@ -7,8 +7,8 @@
 // clang-format on
 #pragma once
 
-#include <host_ir/container.h>
-#include <optimization_pass.h>
```

Comment:
This file is moved to avoid ambiguity between optimization_pass.h and host_ir/pass/optimization_pass.h